src/poh_recorder.rs

```diff
@@ -10,13 +10,21 @@ use std::sync::mpsc::Sender;
 use std::sync::{Arc, Mutex};
 use transaction::Transaction;
 
+pub trait PohRecorder {
```

Comment:
I don't like seeing a trait here or the notion that a PohRecorder might be aware of what a leader or validator is. Instead, how about a boolean state variable that tells PohRecorder how it should behave differentl. The leader would set that bit and the validator wouldn't. Perhaps it could even be updated dynamically so that he object wouldn't need to be dropped to transition.

Also, using the trait has that downstream side-effect of needing to tag the trait with `Send + `static`. That goes away if you simply move the PohRecorder into the service's thread.